### PR TITLE
need feedback on designing interactor

### PR DESCRIPTION
**Pull request by WujiMikeChen**

### CODE REVIEW

File: src/use_case/recipes_getter/GetRecipeInteractor.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    public void execute(GetRecipeInputData getRecipeInputData) {
+        ArrayList<String> ingredients = getRecipeInputData.getIngredients();
+        int number = getRecipeInputData.getNumber();
+        if (!ingredients.isEmpty() && number > 0 && number < 100) {
+            new GetRecipeOutputData(getRecipeDataAccessObject.getResults(ingredients, number));
+            getRecipeInputPresenter.prepareSuccessView();
+        }
+        else {
+            getRecipeInputPresenter.prepareFailView("Check your list of ingredients and number of recipes");
+        }
```
Comment (cher-ning) [inline src/use_case/recipes_getter/GetRecipeInteractor.java:24]: consider making comments for non-trivial code

File: src/use_case/recipes_getter/GetRecipeInteractor.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    public GetRecipeInteractor(GetRecipeDataAccessInterface getRecipeDataAccessObject, GetRecipeOutputBoundary getRecipeInputPresenter) {
+        this.getRecipeDataAccessObject = getRecipeDataAccessObject;
+        this.getRecipeInputPresenter = getRecipeInputPresenter;
+    }
+    @Override
+    public void execute(GetRecipeInputData getRecipeInputData) {
+        ArrayList<String> ingredients = getRecipeInputData.getIngredients();
+        int number = getRecipeInputData.getNumber();
+        if (!ingredients.isEmpty() && number > 0 && number < 100) {
+            new GetRecipeOutputData(getRecipeDataAccessObject.getResults(ingredients, number));
```
Comment (lexazhs) [inline src/use_case/recipes_getter/GetRecipeInteractor.java:19]: when implement GetRecipePresenter, switch to the Result View to show recipes

File: src/use_case/recipes_getter/GetRecipeInteractor.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        if (!ingredients.isEmpty() && number > 0 && number < 100) {
+            GetRecipeOutputData recipes = new GetRecipeOutputData(getRecipeDataAccessObject.getResults(ingredients, number));
+            getRecipeInputPresenter.prepareSuccessView(recipes);
+        }
+        else if (number < 0 | number > 100){
+            getRecipeInputPresenter.prepareFailView("The accepted range for number recipes to return is 0-100");
+        }
+        else {
+            getRecipeInputPresenter.prepareFailView("The ingredient list is empty");
+        }
```
Comment (lexazhs) [inline src/use_case/recipes_getter/GetRecipeInteractor.java:31]: consider adding one more fail view case when there is a typo in their ingredient list

Comment (WujiMikeChen) [reply src/use_case/recipes_getter/GetRecipeInteractor.java:31]: too much work to implement

File: src/interface_adapter/recipes_getter/GetRecipesPresenter.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    }
+    @Override
+    public void prepareSuccessView(GetRecipeOutputData results) {
+        GetRecipesState getRecipesState = getRecipesViewModel.getState();
+        getRecipesState.setRecipes(results.ReturnRecipe());
+        this.getRecipesViewModel.setState(getRecipesState);
+        this.getRecipesViewModel.firePropertyChanged();
+
+        this.viewManagerModel.setActiveView(getRecipesViewModel.getViewName());
+        this.viewManagerModel.firePropertyChanged();
```
Comment (lexazhs) [inline src/interface_adapter/recipes_getter/GetRecipesPresenter.java:30]: consider creating a package called 'Recipe Result' or something like that, and add 'ResultState' and 'ResultViewModel' under that package. Success cases should switch to Result view. Consider using 'LoginPresenter' from week5 coding as a reference.

File: src/interface_adapter/recipe_information_getter/RecipeInfoPresenter.java [hunk 5]
```diff
@@ -1,4 +1,41 @@
...
+    private ViewManagerModel viewManagerModel;
+
+    public RecipeInfoPresenter(RecipeInfoViewModel recipeInfoViewModel, ResultInfoViewModel resultInfoViewModel, ViewManagerModel viewManagerModel) {
+        this.recipeInfoViewModel = recipeInfoViewModel;
+        this.resultInfoViewModel = resultInfoViewModel;
+        this.viewManagerModel = viewManagerModel;
+    }
+
+    @Override
+    public void prepareSuccessView(RecipeInfoOutputData recipe) {
```
Comment (lexazhs) [inline src/interface_adapter/recipe_information_getter/RecipeInfoPresenter.java:24]: Consider adding a comment about which view to switch to

**Pull request merged**